### PR TITLE
ui: Run 2 separate test runs oss and ent

### PR DESCRIPTION
**Pull request by johncowen**
We've always worked towards being able to test out all sorts of permutations of configuration of the UI. Right now we have a 'ui running with namespaces enabled' and a 'ui running with namespaces disabled'.

We had already made our tests loop through all different permutations when namespaces where enabled (empty ns, default ns, filled in ns etc). But we had permanently enabled namespace support for all our tests.

We had hoped to be able to dynamically enabled/disable namespaces with the same test run, but it looks like this isn't possible as the ember test run seems to only runs certain code once (routing, initialisers etc) even though you might have hundreds of tests.

Therefore we now have 2 types of test run, one with namespace support enabled (looping through each type of namespace) and one with namespaces completely disabled.

We've added some more makefile targets/package scripts here to ease this, and we are looking to add both test runs into CI.

One benefit is that we should now be able to run both test runs in parallel.

I've added alvin on here as I'm not sure if we need to give the 2 separate reports different names or not in the circle config?

### CODE REVIEW

File: ui-v2/package.json [hunk 1]
```diff
@@ -22,9 +22,12 @@
     "start:consul": "ember serve --proxy=${CONSUL_HTTP_ADDR:-http://localhost:8500} --port=${EMBER_SERVE_PORT:-4200} --live-reload-port=${EMBER_LIVE_RELOAD_PORT:-7020}",
     "start:api": "api-double --dir ./node_modules/@hashicorp/consul-api-double",
     "test": "ember test --test-port=${EMBER_TEST_PORT:-7357}",
+    "test:oss": "CONSUL_NSPACES_ENABLED=0 ember test --test-port=${EMBER_TEST_PORT:-7357}",
```
Comment (kaxcode) [inline ui-v2/package.json]: Does one of these scripts need to have the `CONSUL_NSPACES_ENABLED` set to true?

Comment (johncowen) [reply ui-v2/package.json]: A good question! I should have explained this also.

I tried to touch as little as possible, so the old `yarn run test` script is still uses `CONSUL_NSPACES_ENABLED=true` by default, so basically if you don't set it on the CLI it defaults to `true`.

I'd actually like to chat with you about this and see what you think. I think we should decide which run (oss or ent) we want as default when you run `make test` - I'm almost on the side of it being the ent one, but let's chat about it when we next hook up. Whichever we decide I think if we change it we should do it in another PR.

**Comment by alvin-huang** [comment]
If you split the tests into two jobs as you have done, they will show test results in the two jobs respectively, running in two separate containers so the same filename won't clash. However, it might still be clearer to name the output files differently while keeping them in the same test result directory just in case we rearrange the logic in the future.

File: .circleci/config.yml [hunk 2]
```diff
@@ -491,11 +491,27 @@ jobs:
...
+      EMBER_TEST_REPORT: test-results/report.xml #outputs test report for CircleCI test summary
+    steps:
+      - checkout
+      - restore_cache:
+          key: *YARN_CACHE_KEY
+      - attach_workspace:
+          at: ui-v2
+      - run:
+          working_directory: ui-v2
+          command: make test-ci
```
Comment (alvin-huang) [inline .circleci/config.yml]: this should be `make test-ci-oss`

Comment (johncowen) [reply .circleci/config.yml]: Oh :doh: great spot Alvin, thanks for that. I just changed this and changed the names of the reports also 👍

**Comment by johncowen** [comment]
I'm gonna merge this down as it will be handy for another bugfix I'm testing out

**Pull request merged**